### PR TITLE
fix(web): open GitHub pull requests in browser when loading fails

### PR DESCRIPTION
**Pull request by t3dotgg**
GitHub API errors can leave a failed pull request view with only Retry, even though the pull request is still available on GitHub.

The error view now includes an Open on GitHub link built from the saved project host and pull request reference, with no extra API request. The shared detail panel covers pull request pages and thread tabs in web and desktop. Error states for GitLab, Bitbucket, Azure DevOps, and unknown providers are unchanged.

## Screenshots

Before

Before: the failed pull request view only shows Retry

After

After: the failed pull request view shows Retry and Open on GitHub

## Checks

- Focused unit tests
- Web typecheck
- Targeted lint and format
- Real browser verification with a local GitHub rate-limit fixture

Made with GPT-5.6 Sol using Codex.

---

> [!NOTE]
> **Low Risk**
> Additive UI fallback with validated URL construction and standard external-link attributes; no API or auth changes.
> 
> **Overview**
> When pull request detail fails to load (e.g. GitHub API errors or rate limits), users are no longer stuck with only **Retry**. The error empty state can show **Open on GitHub** next to retry when a safe browser URL can be derived locally.
> 
> A new `gitHubPullRequestBrowserUrl` helper builds that link from the project’s GitHub `repositoryIdentity` and the open PR reference—using the PR’s `owner/repo` (not the default remote path), honoring GitHub Enterprise HTTP origins without credentials, and falling back to the canonical host for SSH/git remotes. Non-GitHub providers and invalid input yield no link, so other hosts’ error UX is unchanged.
> 
> `PullRequestDetailPanel` wires this into `PullRequestsUnavailableState` on detail load failure. Tests cover URL construction edge cases and the retry/GitHub/action-row combinations; user docs mention **Open on GitHub** in the error view.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit e3462252b7038fe014d69d4154b05a793df22e31. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

> [!NOTE]
> ### Add 'Open on GitHub' fallback action to `PullRequestDetailPanel` error state
> - Adds `gitHubPullRequestBrowserUrl` util in [openPullRequestLink.ts](https://github.com/pingdotgg/t3code/pull/8507/files#diff-953140e01f1329a36fe8f5606c9027215b588e76537644ab9819a9a618d000ae) to build a GitHub PR browser URL from a repository identity, PR number, and repo name; returns `null` for invalid inputs or non-GitHub providers
> - `PullRequestDetailPanel` derives `unavailableGitHubUrl` via `useMemo` from the project's `repositoryIdentity`, and passes it as `gitHubUrl` to `PullRequestsUnavailableState` when `detailQuery.error` is set
> - `PullRequestsUnavailableState` renders an outline "Open on GitHub" button (opens in a new tab with `rel="noopener noreferrer"`) when `gitHubUrl` is provided, alongside the existing optional Retry button
> - Behavioral Change: the error view can now show the new button; `EmptyContent` renders when either `onRetry` or `gitHubUrl` is present
>
> 
>
> <sup><a href="https://app.macroscope.com">Macroscope</a> summarized e346225.</sup>
> 
>

### CODE REVIEW

File: apps/web/src/lib/openPullRequestLink.ts [hunk 1]
```diff
@@ -53,6 +54,132 @@ export async function openPullRequestLink(
...
+  readonly origin: string;
+  readonly path: readonly string[];
+}
+
+function remoteLocationOf(identity: RepositoryIdentity): RemoteLocation | null {
+  try {
+    const url = new URL(identity.locator.remoteUrl.trim());
+    if (url.protocol === "http:" || url.protocol === "https:") {
+      return {
+        origin: url.origin,
```
[the remark this answers is not in the document]
Comment (t3dotgg) [reply apps/web/src/lib/openPullRequestLink.ts]: > [!NOTE]
> 🤖 **GPT-5.6 Sol responding on behalf of Theo**

This is a false positive for the current call path. `buildRepositoryIdentity` in `RepositoryIdentityResolver.ts` keeps every remote pathname segment in `displayName`, and `repositoryIdentityOf` in `PullRequestService.ts` uses that full value for GitLab. For `https://example.com/gitlab/group/repo.git`, the reference is `gitlab/group/repo`, so `pullRequestBrowserUrl` produces `/gitlab/group/repo/-/merge_requests/N`. The prefix is kept exactly once. Adding the remote pathname here would duplicate `/gitlab`.

**Comment by t3dotgg** [comment]
> [!NOTE]
> 🤖 **GPT-5.6 Sol responding on behalf of Theo**

@macroscope-app review

Please re-evaluate the latest commit and the [resolved GitLab prefix finding](https://github.com/pingdotgg/t3code/pull/8507#discussion_r3878867575). `reference.repository` already contains `gitlab/group/repo`, so the helper preserves the install prefix exactly once. No code changed for this dismissal. Please run a fresh correctness and approvability evaluation.

File: apps/web/src/lib/openPullRequestLink.ts [hunk 2]
```diff
@@ -53,6 +54,42 @@ export async function openPullRequestLink(
...
+  let origin: string | null = null;
+  try {
+    const remoteUrl = new URL(identity.locator.remoteUrl.trim());
+    if (remoteUrl.protocol === "http:" || remoteUrl.protocol === "https:") {
+      origin = remoteUrl.origin;
+    }
+  } catch {
+    // SCP-style remotes are read from their normalized identity below.
+  }
+  const hostname = identity.canonicalKey.split("/")[0];
```
[the remark this answers is not in the document]
Comment (t3dotgg) [reply apps/web/src/lib/openPullRequestLink.ts:81]: > [!NOTE]
> 🤖 **GPT-5.6 Sol responding on behalf of Theo**

This is a false positive. `RepositoryIdentity` in `packages/contracts/src/environment.ts` requires `canonicalKey` at runtime, not only in its TypeScript type. `loadShell` in `apps/web/src/connection/storage.ts` decodes persisted snapshots through `OrchestrationShellSnapshot`, and `packages/client-runtime/src/state/shell.ts` discards a cache that fails that decode before `useProjects` can read it. A runtime `Schema.is(RepositoryIdentity)` check also rejects the reported shape without `canonicalKey`. Live identities from `buildRepositoryIdentity` always include it. An absent or null identity is already handled by the provider check, which returns `null` and leaves Retry available.

**Comment by t3dotgg** [comment]
> [!NOTE]
> 🤖 **GPT-5.6 Sol responding on behalf of Theo**

@macroscope-app review

Please re-evaluate the current GitHub-only head and the [resolved `canonicalKey` finding](https://github.com/pingdotgg/t3code/pull/8507#discussion_r3879068164). The `RepositoryIdentity` runtime schema requires `canonicalKey`, and persisted `OrchestrationShellSnapshot` values are decoded before `useProjects` can read them. The reported object without that field is rejected instead of reaching the helper. No code changed for this dismissal. Please run a fresh correctness and approvability evaluation.
